pkg/zsio/detector/peeker.go

```diff
@@ -0,0 +1,52 @@
+package detector
+
+import (
+	"io"
+)
+
+type Peeker struct {
+	io.Reader
+	buffer []byte
+}
+
+func NewPeeker(r io.Reader, n int) (*Peeker, error) {
+	b := make([]byte, n)
+	cc := 0
+	for cc < n {
+		n, err := r.Read(b[cc:])
+		cc += n
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
```

Comment:
If this is a slow drip stream from stdin, it could take a while to fill the 32K buffer.

Reply:
I thought of that but wasn't worried for a v1 of this feature.

Reply:
Fair